src/ir/opt/cfold.rs

```diff
@@ -244,6 +244,29 @@ pub fn fold_cache(node: &Term, cache: &mut TermCache<Term>) -> Term {
                     b.width() + w,
                 ))))
             }),
+            Op::Map(op) => match (get(0).as_array_opt(), get(1).as_array_opt()) {
+                (Some(a), Some(b)) => {
+                    // TODO: extend for n-ary arrays
+                    let mut res = a.clone();
+                    let mut merge = ArrayMerge::new(a.clone(), b.clone());
```

Comment:
Cloning is unnecessary here. ArrayMerge is just reading values (and the default value) out of `a` and `b`, which means it should be able to take a shared reference to each.